### ISSUE TITLE
Duplicate installs, and update page is showing the wrong diff

### ISSUE THREAD

**Issue by joeytwiddle** [Tampermonkey/tampermonkey#1448]
## Expected Behavior

I have been working on my script "Scroll Everywhere [fork]". Sometimes I edit the file in TamperMonkey for testing. This means it sometimes syncs, via Google Drive, between Chrome and Firefox.

But after some work, I also push an update to GitHub and Greasyfork, with a new version number, and usually the new version number causes TamperMonkey to download the update. (This workflow has worked well in the past.)

## Actual Behavior

However, two odd things keep happening:

1. TamperMonkey keeps installing a second version! I try uninstalling the older version, but after a while it comes back again!

![Screenshot_20220317_095004](https://user-images.githubusercontent.com/911799/158720569-019ddef4-9680-402d-a4c8-b5fce82b6f01.png)

2. TamperMonkey keeps prompting me update my "WikiIndent" script (which does NOT need an update). Strangely, everything in the prompt matches WikiIndent, except for the red content in the diff, which is the code from "Scroll Eveywhere [fork]".

![Screenshot_20220316_142724](https://user-images.githubusercontent.com/911799/158720949-6e2f5b2a-5ba3-49d5-af08-18ebad2e2d0a.png)

So something odd is happening, possibly related to Google Drive sync.

This is new behaviour, so it might be related to a recent update to TamperMonkey.

I can survive for now. This isn't too troublesome. But I thought you should know.

## Additional

I tried deleting both of the "Scroll Everywhere [fork]" scripts (and "WikiIndent" too).

Then I went to the Greasyfork page to install it again fresh. But the button on Greasyfork offers to "Reinstall" instead of "Install". And when I do, the Tampermonkey page shows a Reinstall page (which I Canceled). Tampermonkey seems to think the script is still installed! However, if I check in the Dashboard, the script is not listed there.

So I restarted Chrome. All of the scripts I deleted are back!

I will try to reset the Sync. (Unfortunately, if I "Run Sync" or "Reset Sync", then the "Please wait..." popup spins forever.)

After that reset, and then restarting Chrome, "Scroll Everwhere [fork]" is not listed in the Dashboard, and is not running, but the Greasyfork page still says "Reinstall" instead of "Install". The same for "WikiIndent".

## Specifications

- Google Chrome: 99.0.4844.51
- TM: 4.14
- OS: Manjaro Linux

## Script

https://greasyfork.org/en/scripts/399820-scroll-everywhere-fork/code

```js
// ==UserScript==
// @name            Scroll Everywhere [fork]
// @description     Scroll entire page smoothly with long left-click and drag.
// @author          tumpio
// @contributor     joeytwiddle
// @namespace       tumpio@sci.fi
// @homepageURL     https://openuserjs.org/scripts/tumpio/Scroll_Everywhere
// @supportURL      https://github.com/tumpio/gmscripts
// @icon            https://raw.githubusercontent.com/tumpio/gmscripts/master/Scroll_Everywhere/large.png
// @include         *
// @grant           GM_addStyle
// @run-at          document-body
// @version         0.3n
// @license         MIT
// ==/UserScript==
```

**Comment by derjanb** [comment]
The only way this could happen is a script UUID mess up. Can you please check the UUIDs of both script at bottom of their script settings tab? The scripts themselves work fine?

![image](https://user-images.githubusercontent.com/767504/158955317-09a6e0bd-7b9d-406d-a6e6-b4e5167ecd9b.png)

**Comment by joeytwiddle** [comment]
I managed to get rid of the WikiIndent crossover issue, so I'm afraid I cannot check if the UUID is the same. But based on your comment, I guess it was.

For the duplicated "Scroll Everywhere [fork]" scripts, the two scripts have different UUID. One small difference between them is that one has the old Grasemonkey URL, which redirects to the new one:

https://greasyfork.org/scripts/399820-scroll-everywhere/code/Scroll%20Everywhere.user.js
https://greasyfork.org/scripts/399820-scroll-everywhere-fork/code/Scroll%20Everywhere%20%5Bfork%5D.user.js

I'm not sure how it got the old URL. Presumably from old Sync data.

Both scripts run ok.

I tried deleting both scripts, but when I go to Greasyfork to install it again, I still get this strange message "Reinstall version 0.3o" when it should say "Install". Why does TM think the script is installed, if it's not in the Dashboard list?

Anyway what I have tried just now is to export my TM data, do a factory reset, and then import my TM data.

I will also reset my sync, and wait a day, to see if the sync brings back the duplicate.

**Comment by joeytwiddle** [comment]
lol, good morning!

![Screenshot_20220322_131440](https://user-images.githubusercontent.com/911799/159413528-e325861a-f767-4d23-be1d-f015a7f2259a.png)

So I think it must be something wrong with my Google Drive Sync.

Unfortunately when I hit "Settings > Userscript Sync > Reset" it just keeps spinning.

Is there any way to verify that the sync has actually been cleared? Or can I do it manually on Google Drive?

**Comment by joeytwiddle** [comment]
What I did: Factory reset TamperMonkey in both browsers, cleared the Google Drive, re-imported from zip (which doesn't include "Scroll Everywhere [fork]") then installed "Scroll Everywhere [fork]" from GreasyFork. I enabled Google Drive Sync for Chrome browser only.

What happened? After 3 days, the same duplication as noted before, but now with 4 of them! (Two copies of the one at the [old address](https://greasyfork.org/scripts/399820-scroll-everywhere/code/Scroll%20Everywhere.user.js), each script with a different ID.)

What I am trying now: On GreasyFork site, I am renaming the script from "Scroll Everywhere [fork]" back to "Scroll Everywhere". I didn't want to do that, but let's see if it helps...
